File: csharp/AdventOfCode.UnitTests/Puzzles/Day07Should.cs
using System.Collections.Generic;
using AdventOfCode.Core.Puzzles;
using Xunit;

namespace AdventOfCode.UnitTests.Puzzles;

public class Day07Should
{
    [Fact]
    public void SolvePart1Test()
    {
        // ARRANGE
        var day7 = new Day07();

        // ACT
        var result = day7.SolvePart1("Puzzles/Day07TestInput.txt");

        // ASSERT
        Assert.NotNull(result);
        Assert.Equal(37, result);
    }

    [Fact]
    public void SolvePart1()
    {
        // ARRANGE
        var day7 = new Day07();

        // ACT
        var result = day7.SolvePart1("Puzzles/Day07Input.txt");

        // ASSERT
        Assert.NotNull(result);
        Assert.Equal(356990, result);
    }
    
    [Fact]
    public void SolvePart2Test()
    {
        // ARRANGE
        var day7 = new Day07();

        // ACT
        var result = day7.SolvePart2("Puzzles/Day07TestInput.txt");

        // ASSERT
        Assert.NotNull(result);
        Assert.Equal(168, result);
        
    }

    [Fact]
    public void SolvePart2()
    {
        // ARRANGE
        var day7 = new Day07();

        // ACT
        var result = day7.SolvePart2("Puzzles/Day07Input.txt");

        // ASSERT
        Assert.NotNull(result);
        Assert.Equal(101267361, result);
    }
}

File: fsharp/AdventOfCode.UnitTests/Puzzles/Day07Should.fs
module AdventOfCode.UnitTests.Puzzles.Day07Should

open AdventureOfCode.Puzzles
open Xunit

[<Fact>]
let ``Return the correct calculation for the example of part 1`` () =
    let result = Day07.solvePart1 "Puzzles/Day07TestInput.txt"
    Assert.NotNull(result)
    Assert.Equal(66, result)
    
    
[<Fact>]
let ``Return the correct calculation for part 1`` () =
    let result = Day07.solvePart1 "Puzzles/Day07Input.txt"
    Assert.NotNull(result)
    Assert.Equal(13933, result)
    
[<Fact>]
let ``Return the correct calculation for the example of part 2`` () =
    let result = Day07.solvePart2 "Puzzles/Day07TestInput.txt"
    Assert.NotNull(result)
    Assert.Equal(168, result)
    
    
[<Fact>]
let ``Return the correct calculation for part 2`` () =
    let result = Day07.solvePart2 "Puzzles/Day07Input.txt"
    Assert.NotNull(result)
    Assert.Equal(101268110, result)